pystac_client/mixins.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional, Dict, Any
+import pystac
+from pystac_client.exceptions import APIError
+from pystac_client.conformance import ConformanceClasses
+from pystac_client.stac_api_io import StacApiIO
+
+QUERYABLES_REL = "http://www.opengis.net/def/rel/ogc/1.0/queryables"
+QUERYABLES_ENDPOINT = "/queryables"
+
+
+class StacAPIObject(pystac.STACObject):
+    _stac_io: Optional[StacApiIO]
```

Comment:
In earlier commits I had this as a Protocol, and was using that as the type for `self` in the `Mixin`. I like this pattern better because it saves me from duplicating whole method definitions from `pystac.STACObject`. 

Inspired by: https://stackoverflow.com/a/70907644/4021797